11 спринт (1я попытка)

ValentinPside opened the pull request:
Совсем запутался, не понимаю, что не так и почему не работает

ValentinPside committed:
11 спринт (1я попытка)

faritowich commented on app/src/main/java/com/example/playlistmaker/DateUtils.kt:
```diff
@@ -0,0 +1,10 @@
+object DateUtils {
+    fun formatTime(trackTime: Long): String {
+        return SimpleDateFormat("mm:ss", Locale.getDefault()).format(trackTime)
+    }
```
Можно вместо этого использовать экстеншн функцию:

`fun Long.formatTime(): String = SimpleDateFormat("mm:ss", Locale.getDefault()).format(this)`

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -57,6 +80,53 @@ class SearchActivity : AppCompatActivity() {
             val view: View? = this.currentFocus
             inputMethodManager?.hideSoftInputFromWindow(view?.windowToken, 0)
         }
+         fun sendRequestToServer() {
```
private

faritowich commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -87,4 +87,81 @@
+            app:cornerRadius="54dp"
+            style="@style/primary_text"/>
+
+    </LinearLayout>
```
Лучше реализовать показ заглушек не разными лейаутами communication_problem_placeholder и nothing_found_placeholder, а объединить в один, а изображение, текст и видимость кнопки "обновить" задавать в активити. Можно создать в активити отдельный метод, в аргументы которого передается энам статуса поиска (этот энам надо создать и сделать в нем статусы не найдено, нет сети, успех) и в зависимости от переданного энама задавать картинку, сеттить  текст ошибки, видимость кнопки Обновить и т,д.. Тогда будет меньше загружен xml-файл, а также упростится логика работы в активити

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -6,28 +6,51 @@ import android.os.Bundle
+    private lateinit var searchTextField: EditText
+
     private companion object {
         const val EDIT_TEXT = "EDIT_TEXT"
```
private

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -6,28 +6,51 @@ import android.os.Bundle
+        const val ITUNES_BASE_URL = "https://itunes.apple.com"
     }
+
     var searchLineText : String? = null
```
private

ValentinPside committed:
2я попытка

ValentinPside commented:
Перезалил проект целиком, проверил, чтоб не крашился. В целом вопрос тот же, что на эмуляторе, что на реальном устройстве не выводится список треков, не могу найти ошибку самостоятельно

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -57,6 +80,53 @@ class SearchActivity : AppCompatActivity() {
+        }
+        communicationProblemButton.setOnClickListener {
+            sendRequestToServer()
+        }
```
Дублирование кода

ValentinPside committed:
3я попытка

1)Исправлены модификаторы доступа у методов
2)Использована экстеншн функция для форматирования времени
3)Исправлен actionId, который вызывает поиск треков

ValentinPside committed:
Перезалил проект полностью, так при проверке возникли ошибки компиляции

Попутно исправлено несколько мелочей, связанных с картинками и их отображением при светлой и тёмной теме

ValentinPside commented:
![Скрины](https://github.com/ValentinPside/Playlist-Maker/assets/91324342/8e7450d5-2125-472b-a426-547dc9cf3339)
Дополнительно проверил на реальном устройстве

faritowich commented on app/src/main/java/com/example/playlistmaker/TrackViewHolder.kt:
```diff
@@ -1,24 +1,30 @@
     private val bandName: TextView = itemView.findViewById(R.id.band_name)
     private val trackTime: TextView = itemView.findViewById(R.id.track_time)
 
+    fun Long.formatTime(): String = SimpleDateFormat("mm:ss", Locale.getDefault()).format(this)
```
Хорошая идея написать экстеншн, но лучше его держать не во вьюхолдере, а вынести отдельно. Например, создать для таких полезных штук файл Utils
